portal/portal-service-impl/impl/src/java/org/sakaiproject/portal/service/PortalServiceImpl.java

```diff
@@ -821,6 +821,10 @@ public List<String> getRecentSites() {
 	@Transactional
 	public void addRecentSite(String siteId) {
 
+		if (siteId.equals(SiteService.SITE_ERROR)) {
```

Comment:
Operands should be reversed to provide NPE protection:

```suggestion
		if (SiteService.SITE_ERROR.equals(siteId)) {
```